bin/src/command/mod.rs

```diff
@@ -474,41 +474,44 @@ impl CommandServer {
 
         let tx = command_tx.clone();
 
-        let workers: Vec<Worker> = workers
-            .iter()
-            .filter_map(move |serialized| {
-                if serialized.run_state == RunState::Stopped
-                    || serialized.run_state == RunState::Stopping
-                {
-                    return None;
-                }
+        let mut workers: Vec<Worker> = Vec::new();
 
-                let (worker_tx, worker_rx) = channel(10000);
-                let sender = Some(worker_tx);
+        for serialized in serialized_workers.iter() {
+            if serialized.run_state == RunState::Stopped
+                || serialized.run_state == RunState::Stopping
+            {
+                continue;
+            }
 
-                debug!("deserializing worker: {:?}", serialized);
-                let stream = Async::new(unsafe { UnixStream::from_raw_fd(serialized.fd) }).unwrap();
+            let (worker_tx, worker_rx) = channel(10000);
+            let sender = Some(worker_tx);
 
-                let id = serialized.id;
-                let command_tx = tx.clone();
-                //async fn worker(id: u32, sock: Async<UnixStream>, tx: Sender<CommandMessage>, rx: Receiver<()>) -> std::io::Result<()> {
-                smol::spawn(async move {
-                    worker_loop(id, stream, command_tx, worker_rx).await;
-                })
-                .detach();
+            debug!("deserializing worker: {:?}", serialized);
+            let stream = Async::new(unsafe { UnixStream::from_raw_fd(serialized.fd) }).unwrap();
```

Comment:
At least, it should be an expect, but I wonder if it will be better to propagate the error instead

Reply:
YES. An error on `Async::new()` is unlikely but would definitely be fatal to worker creation.